File: LeetCode Solutions/March 2022 LeetCoding Challenge/Broken Calculator/Broken Calculator.c


int brokenCalc(int startValue, int target){
    int result = 0;
    while(startValue < target){
        result++;
        if(target % 2 > 0){
            target++;
        } else{
            target /= 2;
        }
    }
    return result + startValue - target;
}

File: LeetCode Solutions/March 2022 LeetCoding Challenge/Broken Calculator/Broken Calculator.py
class Solution:
    def brokenCalc(self, startValue: int, target: int) -> int:
        result = 0
        while(startValue < target):
            result += 1
            if(target % 2):
                target += 1
            else:
                target //= 2
        return result + startValue - target